server/services/v1/query_runner.go

```diff
@@ -539,19 +539,19 @@ func (runner *StreamingQueryRunner) ReadOnly(ctx context.Context, tenant *metada
 		}
 
 		var last []byte
-		if last, err = runner.iterateOnKvStore(ctx, tx, options); err == nil {
-			_ = tx.Commit(ctx)
-			return &Response{}, ctx, nil
-		}
-		_ = tx.Rollback(ctx)
+		last, err = runner.iterateOnKvStore(ctx, tx, options)
+		_ = tx.End(ctx)
```

Comment:
Why not just tx.Rollback() ?

Reply:
It's not obvious what End does from the name.

Reply:
So I want to avoid calling Rollback, it has readability concerns. I am open to naming though.